src/compiler/checker.ts

```diff
@@ -25576,7 +25585,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         );
     }
 
-    function clearCachedInferences(inferences: InferenceInfo[]) {
+    function clearCachedInferences(context: InferenceContext | undefined, inferences: InferenceInfo[]) {
+        if (context) {
+            context.nonFixingMapper.instantiations = undefined;
+        }

```

Comment:
`nonFixingMapper` isn't idempotent so whenever inferences are cleared it's instantiation cache has to be cleared too